internal/buffer/save.go

```diff
@@ -153,26 +197,27 @@ func (b *Buffer) saveToFile(filename string, withSudo bool, autoSave bool) error
 		}
 	}
 
+	filename, _ = util.ReplaceHome(filename)
```

Comment:
One more thought: it doesn't seem quite ok that we just ignore the error here...

Reply:
I assume we should treat `absFilename, _ := filepath.Abs(filename)` then the same and care about the error.

Furthermore...
```
	// Update the last time this file was updated after saving
	defer func() {
		b.ModTime, _ = util.GetModTime(filename)
		err = b.Serialize()
	}()
```
...should be moved behind the actual successful save request or even the end of the `saveToFile()` function. Because otherwise the modification time will be updated and the buffer serialized without the safe and successful save.